src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombiner.java

```diff
@@ -80,16 +78,15 @@ private List<ScoreDoc> getCombinedScoreDocs(
         final CompoundTopDocs compoundQueryTopDocs,
         final Map<Integer, Float> combinedNormalizedScoresByDocId,
         final List<Integer> sortedScores,
-        final int maxHits
+        final long maxHits
     ) {
-        ScoreDoc[] finalScoreDocs = new ScoreDoc[maxHits];
-
+        List<ScoreDoc> scoreDocs = new ArrayList<>();
         int shardId = compoundQueryTopDocs.getScoreDocs().get(0).shardIndex;
         for (int j = 0; j < maxHits && j < sortedScores.size(); j++) {
```

Comment:
shouldn't j be long too?

Reply:
We have 2 conditions here, j<maxHits and sortedScores.size().  The sortedScores.size() condition will always return int. Therefore, It will be for j to be int.